pkg/logs/sender/sender.go

```diff
@@ -89,3 +98,117 @@ func (s *Sender) send(payload *message.Message) {
 	}
 	s.outputChan <- payload
 }
+
+// BatchSender is responsible for sending logs to different destinations in batches
+type BatchSender struct {
+	inputChan     chan *message.Message
+	outputChan    chan *message.Message
+	destinations  *client.Destinations
+	done          chan struct{}
+	batchTimeout  time.Duration
+	messageBuffer *MessageBuffer
+}
+
+// NewBatchSender returns an new BatchSender.
+func NewBatchSender(inputChan, outputChan chan *message.Message, destinations *client.Destinations) *BatchSender {
+	return &BatchSender{
+		inputChan:     inputChan,
+		outputChan:    outputChan,
+		destinations:  destinations,
+		done:          make(chan struct{}),
+		batchTimeout:  5 * time.Second,               // TODO Change this to paramter
+		messageBuffer: NewMessageBuffer(20, 2000000), // TODO change this to paramter
+	}
+}
+
+// Start starts the BatchSender
+func (b *BatchSender) Start() {
+	go b.run()
+}
+
+// Stop stops the BatchSender,
+// this call blocks until inputChan is flushed
+func (b *BatchSender) Stop() {
+	close(b.inputChan)
+	<-b.done
+}
+
+// run lets the BatchSender send messages.
+func (b *BatchSender) run() {
+	flushTimer := time.NewTimer(b.batchTimeout)
+	defer func() {
+		flushTimer.Stop()
+		b.done <- struct{}{}
+	}()
+
+	for {
+		select {
+		case payload, isOpen := <-b.inputChan:
+			if !isOpen {
+				// inputChan has been closed, no more payload are expected
+				// TODO remember to flush your buffer!!
+				return
+			}
+			sendNow, appendAgain := b.messageBuffer.tryAppendToBuffer(payload)
```

Comment:
I think append to buffer should not return any logic to know when to send, when to append, this logic should be done inside this method.